tools/clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -7771,6 +7771,8 @@ def err_hlsl_unsupported_for_version_lower : Error<
    "%0 is only allowed for HLSL %1 and lower.">;
 def err_hlsl_unsupported_keyword_for_min_precision : Error<
    "%0 is only supported with -enable-16bit-types option">;
+def err_hlsl_derivatives_intrinsic_in_wrong_shader_kind : Error<
```

Comment:
Can we instead make this a warning that is `DefaultError`? Since we still need to have the validator catch these cases this would allow developers to disable the error (or make it a warning) if it interferes with their code in the ways that @tex3d has mentioned.

Reply:
Updated.